### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by schneiderman805**
- We created a table to hold our products 

- we created a feature file to test the menu

- we created a root path

- we created an index.html
pt:
https://www.pivotaltracker.com/n/projects/2227584
download

### CODE REVIEW

File: app/controllers/welcome_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class WelcomeController < ApplicationController
+    def index
+
```
Comment (faraznaeem) [inline app/controllers/welcome_controller.rb]: Why do you have two controllers?

File: app/views/welcome/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+          <li>green salad</li>
+          <li>fresh lettuce, tomato and cheese</li>
+          <li>50</li>
+          <br>
+          <li>pasta carbonara</li>
+          <li>pasta with some delicious cheese sauce</li>
+          <li>102</li>
+          <br>
+          <li>ice cream</li>
+          <li>vanilla with chocolate sauce</li>
```
Comment (faraznaeem) [inline app/views/welcome/index.html.erb]: You have hardcoded the values, that is not what we want, we want to bring the values from the db

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-end
+    get 'welcome/index'
```
Comment (faraznaeem) [inline config/routes.rb]: you need to use resources

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
...
+
+    root 'welcome#index'
```
Comment (faraznaeem) [inline config/routes.rb]: This is not the way we write root controller, please have another look at the routing demo

File: spec/factories/products.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :product do
+
```
Comment (faraznaeem) [inline spec/factories/products.rb]: Which model do you want to use, products or meals? choose one

File: spec/models/product_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Product, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (faraznaeem) [inline spec/models/product_spec.rb]: no test for the model you need to have those

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/controllers/welcome_controller.rb (hunk 1 above)
Comment (HannaSwDn) [reply app/controllers/welcome_controller.rb]: We got rid of the welcome controller

Comment (HannaSwDn) [reply app/controllers/welcome_controller.rb]: We accidentally added it when we didn't need to

File: app/views/welcome/index.html.erb (hunk 2 above)
Comment (HannaSwDn) [reply app/views/welcome/index.html.erb]: Removed the hardcoding and created a variable to store all products in. We now output the products to the HAML file from the database

File: config/routes.rb (hunk 3 above)
Comment (HannaSwDn) [reply config/routes.rb]: We are now using resources

File: config/routes.rb (hunk 4 above)
Comment (HannaSwDn) [reply config/routes.rb]: We have re-written it now

File: spec/factories/products.rb (hunk 5 above)
Comment (HannaSwDn) [reply spec/factories/products.rb]: We have decided to use products

File: spec/models/product_spec.rb (hunk 6 above)
Comment (HannaSwDn) [reply spec/models/product_spec.rb]: We have implemented tests now

File: config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-end
+    resources :products
```
Comment (faraznaeem) [inline config/routes.rb]: restrict the resources

File: spec/factories/meals.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+    factory :meal do |f|
+        f.starter 'salad'
+        f.main 'pizza'
+        f.dessert 'ice cream'
+    end
+end
```
Comment (faraznaeem) [inline spec/factories/meals.rb]: delete this file if its not in use

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: spec/factories/meals.rb (hunk 8 above)
Comment (HannaSwDn) [reply spec/factories/meals.rb]: Deleted!

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**